charts/opentelemetry-collector/templates/cr-collector.yaml

```diff
@@ -0,0 +1,60 @@
+{{- if .Values.defineCustomResource -}}
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: {{ include "opentelemetry-collector.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "opentelemetry-collector.labels" . | nindent 4 }}
+  {{- if .Values.annotations }}
+  annotations:
+    {{- range $key, $value := .Values.annotations }}
+      {{- printf "%s: %s" $key (tpl $value $ | quote) | nindent 4 }}
+      {{- end }}
+  {{- end }}
+spec:
+  mode: {{ .Values.mode }}
```

Comment:
To be fully configurable we need to allow `mode` to be `sidecar`.  We should fail in `NOTES.txt` if `mode=sidecar` and `defineCustomResource=false`